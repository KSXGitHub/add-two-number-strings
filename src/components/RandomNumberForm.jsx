import React from 'react'
import Paper from 'material-ui/Paper'
import {Card, CardActions, CardHeader, CardText} from 'material-ui/Card'
import Slider from 'material-ui/Slider'
import Checkbox from 'material-ui/Checkbox'
import RandomNumber from './RandomNumber.jsx'

export default class RandomNumberForm extends React.Component {
  constructor (props) {
    super(props)

    const {
      begin = 0,
      end = 13,
      uppercase = true,
      defaultContent = <i>(Empty)</i>,
      init
    } = props

    this.state = {
      begin,
      end,
      uppercase,
      defaultContent,
      init
    }
  }

  render () {
    const getText = float => (
      (parseFloat(float)
        .toString(16)
        .slice(2)
        .slice(this.state.begin, this.state.end)
      )[this.state.uppercase ? 'toUpperCase' : 'toLowerCase']()
    ) || this.state.defaultContent

    const randomNumber = <RandomNumber
      display={getText}
      init={this.state.init}
      style={{
        label: {
          color: 'gray'
        },
        value: {
          color: 'brown',
          fontFamily: 'monospace',
          fontSize: '1.75em'
        }
      }}
    />

    return <Paper zDepth={1}><Card>
      <CardHeader
        title='Random Number'
        subtitle={randomNumber}
        actAsExpander
        showExpandableButton
      />

      <CardText expandable>
        <CardActions expandable>
          <p>
            <label htmlFor='begin-slider'>Begin</label>
          </p>

          <Slider
            id='begin-slider'
            min={0}
            max={this.state.end}
            step={1}
            value={this.state.begin}
            onChange={(_, begin) => begin < this.state.end && this.setState({begin})}
          />

          <p>
            <label htmlFor='end-slider'>End</label>
          </p>

          <Slider
            id='end-slider'
            min={this.state.begin}
            max={13}
            step={1}
            value={this.state.end}
            onChange={(_, end) => end > this.state.begin && this.setState({end})}
          />

          <Checkbox
            label='Upper Case'
            checked={this.state.uppercase}
            onCheck={(_, uppercase) => this.setState({uppercase})}
            style={{marginBottom: 16}}
            labelPosition='right'
          />
        </CardActions>
      </CardText>
    </Card></Paper>
  }
}
