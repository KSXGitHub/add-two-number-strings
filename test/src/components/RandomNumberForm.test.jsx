import './__lib__/material-ui-mocks.jsx'
import React from 'react'
import renderer from 'react-test-renderer'
import RandomNumberForm from '../../../src/components/RandomNumberForm.jsx'

describe('Snapshot: components/RandomNumberForm.jsx', () => {
  const init = () => 0.123456789

  test('with default display function', () => {
    const tree = renderer.create(<RandomNumberForm init={init} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('with specified display function', () => {
    ; [
      value => JSON.stringify({value}),
      value => Number(value).toString(16).slice(2, 16),
      value => Number(value).toString(36).slice(2, 13),
      value => value * value,
      value => [0, 1, 2, 3].map(x => value * x).join(', ')
    ]
      .map(display => (<RandomNumberForm init={init} display={display} />))
      .map(element => renderer.create(element))
      .map(instance => instance.toJSON())
      .forEach(tree => expect(tree).toMatchSnapshot())
  })
})
