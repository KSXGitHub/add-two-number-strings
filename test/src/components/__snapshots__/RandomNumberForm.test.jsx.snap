// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: components/RandomNumberForm.jsx with default display function 1`] = `
<mocked-component
  zDepth={1}
>
  <div>
    <div
      className="text-field-container"
    >
      <p>
        <label
          htmlFor="begin-slider"
        >
          Begin
        </label>
      </p>
      <mocked-component
        id="begin-slider"
        max={13}
        min={0}
        onChange={[Function]}
        step={1}
        value={0}
      />
      <p>
        <label
          htmlFor="end-slider"
        >
          End
        </label>
      </p>
      <mocked-component
        id="end-slider"
        max={13}
        min={0}
        onChange={[Function]}
        value={13}
      />
      <mocked-component
        checked={true}
        label="UpperCase"
        labelPosition="right"
        onCheck={[Function]}
        style={
          Object {
            "marginBottom": 16,
          }
        }
      />
    </div>
    <div
      className="output-container"
    >
      <mocked-component
        zDepth={0}
      >
        <span
          className="label"
          style={
            Object {
              "color": "gray",
            }
          }
        >
          Random Number: 
        </span>
        <span
          className="value"
          style={
            Object {
              "color": "brown",
              "fontFamily": "monospace",
              "fontSize": "1.75em",
            }
          }
        >
          1F9ADD3739635
        </span>
      </mocked-component>
    </div>
  </div>
</mocked-component>
`;

exports[`Snapshot: components/RandomNumberForm.jsx with specified display function 1`] = `
<mocked-component
  zDepth={1}
>
  <div>
    <div
      className="text-field-container"
    >
      <p>
        <label
          htmlFor="begin-slider"
        >
          Begin
        </label>
      </p>
      <mocked-component
        id="begin-slider"
        max={13}
        min={0}
        onChange={[Function]}
        step={1}
        value={0}
      />
      <p>
        <label
          htmlFor="end-slider"
        >
          End
        </label>
      </p>
      <mocked-component
        id="end-slider"
        max={13}
        min={0}
        onChange={[Function]}
        value={13}
      />
      <mocked-component
        checked={true}
        label="UpperCase"
        labelPosition="right"
        onCheck={[Function]}
        style={
          Object {
            "marginBottom": 16,
          }
        }
      />
    </div>
    <div
      className="output-container"
    >
      <mocked-component
        zDepth={0}
      >
        <span
          className="label"
          style={
            Object {
              "color": "gray",
            }
          }
        >
          Random Number: 
        </span>
        <span
          className="value"
          style={
            Object {
              "color": "brown",
              "fontFamily": "monospace",
              "fontSize": "1.75em",
            }
          }
        >
          1F9ADD3739635
        </span>
      </mocked-component>
    </div>
  </div>
</mocked-component>
`;

exports[`Snapshot: components/RandomNumberForm.jsx with specified display function 2`] = `
<mocked-component
  zDepth={1}
>
  <div>
    <div
      className="text-field-container"
    >
      <p>
        <label
          htmlFor="begin-slider"
        >
          Begin
        </label>
      </p>
      <mocked-component
        id="begin-slider"
        max={13}
        min={0}
        onChange={[Function]}
        step={1}
        value={0}
      />
      <p>
        <label
          htmlFor="end-slider"
        >
          End
        </label>
      </p>
      <mocked-component
        id="end-slider"
        max={13}
        min={0}
        onChange={[Function]}
        value={13}
      />
      <mocked-component
        checked={true}
        label="UpperCase"
        labelPosition="right"
        onCheck={[Function]}
        style={
          Object {
            "marginBottom": 16,
          }
        }
      />
    </div>
    <div
      className="output-container"
    >
      <mocked-component
        zDepth={0}
      >
        <span
          className="label"
          style={
            Object {
              "color": "gray",
            }
          }
        >
          Random Number: 
        </span>
        <span
          className="value"
          style={
            Object {
              "color": "brown",
              "fontFamily": "monospace",
              "fontSize": "1.75em",
            }
          }
        >
          1F9ADD3739635
        </span>
      </mocked-component>
    </div>
  </div>
</mocked-component>
`;

exports[`Snapshot: components/RandomNumberForm.jsx with specified display function 3`] = `
<mocked-component
  zDepth={1}
>
  <div>
    <div
      className="text-field-container"
    >
      <p>
        <label
          htmlFor="begin-slider"
        >
          Begin
        </label>
      </p>
      <mocked-component
        id="begin-slider"
        max={13}
        min={0}
        onChange={[Function]}
        step={1}
        value={0}
      />
      <p>
        <label
          htmlFor="end-slider"
        >
          End
        </label>
      </p>
      <mocked-component
        id="end-slider"
        max={13}
        min={0}
        onChange={[Function]}
        value={13}
      />
      <mocked-component
        checked={true}
        label="UpperCase"
        labelPosition="right"
        onCheck={[Function]}
        style={
          Object {
            "marginBottom": 16,
          }
        }
      />
    </div>
    <div
      className="output-container"
    >
      <mocked-component
        zDepth={0}
      >
        <span
          className="label"
          style={
            Object {
              "color": "gray",
            }
          }
        >
          Random Number: 
        </span>
        <span
          className="value"
          style={
            Object {
              "color": "brown",
              "fontFamily": "monospace",
              "fontSize": "1.75em",
            }
          }
        >
          1F9ADD3739635
        </span>
      </mocked-component>
    </div>
  </div>
</mocked-component>
`;

exports[`Snapshot: components/RandomNumberForm.jsx with specified display function 4`] = `
<mocked-component
  zDepth={1}
>
  <div>
    <div
      className="text-field-container"
    >
      <p>
        <label
          htmlFor="begin-slider"
        >
          Begin
        </label>
      </p>
      <mocked-component
        id="begin-slider"
        max={13}
        min={0}
        onChange={[Function]}
        step={1}
        value={0}
      />
      <p>
        <label
          htmlFor="end-slider"
        >
          End
        </label>
      </p>
      <mocked-component
        id="end-slider"
        max={13}
        min={0}
        onChange={[Function]}
        value={13}
      />
      <mocked-component
        checked={true}
        label="UpperCase"
        labelPosition="right"
        onCheck={[Function]}
        style={
          Object {
            "marginBottom": 16,
          }
        }
      />
    </div>
    <div
      className="output-container"
    >
      <mocked-component
        zDepth={0}
      >
        <span
          className="label"
          style={
            Object {
              "color": "gray",
            }
          }
        >
          Random Number: 
        </span>
        <span
          className="value"
          style={
            Object {
              "color": "brown",
              "fontFamily": "monospace",
              "fontSize": "1.75em",
            }
          }
        >
          1F9ADD3739635
        </span>
      </mocked-component>
    </div>
  </div>
</mocked-component>
`;

exports[`Snapshot: components/RandomNumberForm.jsx with specified display function 5`] = `
<mocked-component
  zDepth={1}
>
  <div>
    <div
      className="text-field-container"
    >
      <p>
        <label
          htmlFor="begin-slider"
        >
          Begin
        </label>
      </p>
      <mocked-component
        id="begin-slider"
        max={13}
        min={0}
        onChange={[Function]}
        step={1}
        value={0}
      />
      <p>
        <label
          htmlFor="end-slider"
        >
          End
        </label>
      </p>
      <mocked-component
        id="end-slider"
        max={13}
        min={0}
        onChange={[Function]}
        value={13}
      />
      <mocked-component
        checked={true}
        label="UpperCase"
        labelPosition="right"
        onCheck={[Function]}
        style={
          Object {
            "marginBottom": 16,
          }
        }
      />
    </div>
    <div
      className="output-container"
    >
      <mocked-component
        zDepth={0}
      >
        <span
          className="label"
          style={
            Object {
              "color": "gray",
            }
          }
        >
          Random Number: 
        </span>
        <span
          className="value"
          style={
            Object {
              "color": "brown",
              "fontFamily": "monospace",
              "fontSize": "1.75em",
            }
          }
        >
          1F9ADD3739635
        </span>
      </mocked-component>
    </div>
  </div>
</mocked-component>
`;
